setup.py

```diff
@@ -68,7 +64,7 @@
         ],
     },
 
-    python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",
+    python_requires=">=3.6",
```

Comment:
I'm sorry I didn't get to comment before this was merged, but while I agree with everything else in this PR, I think this is overly restrictive. There is a difference here in saying what we will support (via classifiers) vs. what users are able to install. 

Unless we know for sure that `twine` will not work with `<=3.5`, we are excluding users from using `twine` and newer features without much reason.

I think this should probably be `>=3.4` and we should continue running CI against 3.4 and 3.5 as "allowed failures" and make this more restrictive once/if they start failing.

Reply:
The patch does require Python 3.6 due to use of f-strings. Removing the `python_requires` and running `tox -e py35` yields this result:

```
ImportError while loading conftest '/Users/jaraco/code/public/pypa/twine/tests/conftest.py'.
tests/conftest.py:5: in <module>
    from twine import settings
twine/settings.py:16: in <module>
    from twine import repository
E     File "/Users/jaraco/code/public/pypa/twine/twine/repository.py", line 111
E       print(f"Registering {package.basefilename}")
E                                                 ^
E   SyntaxError: invalid syntax
ERROR: InvocationError for command /Users/jaraco/code/public/pypa/twine/.tox/py35/bin/coverage run --source twine -m pytest tests (exited with code 4)
```

I'm personally a big fan of f-strings, so I'm slightly disinclined to backport to Python 3.5. I'm more strongly resistant to backporting to Python 3.4, given that it's reached EOL (and twine 1.x still supports it).

In a61a4be, I've employed future-fstrings to keep f-strings, but tests still fail due to a missing hash implementation:

```
ImportError while importing test module '/Users/jaraco/code/public/pypa/twine/tests/test_check.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_check.py:16: in <module>
    from twine.commands import check
twine/commands/check.py:23: in <module>
    from twine.package import PackageFile
twine/package.py:21: in <module>
    from hashlib import blake2b
E   ImportError: cannot import name 'blake2b'
________________________________________
```

I haven't gone farther to determine what other Python 3.6 features might be present.

In light of those concerns, should we invest the energy to continue to backport to 3.4 or 3.5?

Reply:
I see now that we've undone some compatability code in this PR.

I think generally our policy should be not to do this unless we really need to use some feature or library.

I'm not sure if it's worth adding compatability back at this point: the goal should be for us to do less work, not more.

Reply:
I realize this is a small consideration, and squarely in the developer experience realm, but I noted in https://github.com/pypa/twine/pull/437#issuecomment-501080341 that requiring 3.6+ would let us use [type annotations on variables](https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html#variables) (instead of type comments).

Reply:
Given that `python_requires>=3.6` will prevent this version from being installed on Python 3.5 and earlier (pip 9+), _and_ that Twine 1.x will be installed instead, I'm very much in favor of moving forward. Thanks for raising this concern, though. I learned some things in the investigation that I'd missed before.